illumos-utils/src/coreadm.rs

```diff
@@ -0,0 +1,57 @@
+use camino::Utf8PathBuf;
+use std::ffi::OsString;
+use std::os::unix::ffi::OsStringExt;
+use std::process::Command;
+
+#[derive(thiserror::Error, Debug)]
+pub enum CoreAdmError {
+    #[error("Error obtaining or modifying coreadm configuration. core_dir: {core_dir:?}")]
+    Execution { core_dir: Utf8PathBuf },
+
+    #[error("Invalid invocation of coreadm: {0:?} {1:?}")]
+    InvalidCommand(Vec<String>, OsString),
+
+    #[error("coreadm process was terminated by a signal.")]
+    TerminatedBySignal,
+
+    #[error("coreadm invocation exited with unexpected return code {0}")]
+    UnexpectedExitCode(i32),
+
+    #[error("Failed to execute dumpadm process: {0}")]
+    Exec(std::io::Error),
+}
+
+const COREADM: &str = "/usr/bin/coreadm";
+
+pub fn coreadm(core_dir: &Utf8PathBuf) -> Result<(), CoreAdmError> {
+    let mut cmd = Command::new(COREADM);
+    cmd.env_clear();
+
+    // disable per-process core patterns
+    cmd.arg("-d").arg("process");
+
+    // use the global core pattern
+    cmd.arg("-e").arg("global");
+
```

Comment:
I'd suggest also explicitly configuring the core file content here, and enabling core dumps from `setid` processes too.

```
-e global-setid
```

```
-G stack+heap+shm+ism+dism+text+data+rodata+anon+shanon+ctf+symtab
```
(and consider adding `debug` to also include any DWARF data from the executable and libraries)

You could also use `-G default` or `-G default+debug` but the default set may not be stable.